src/streamlink/plugins/vlive.py

```diff
@@ -9,48 +9,60 @@
-        if vinit_req.status_code != 200:
-            raise PluginError('Could not get video init page (HTTP Status {})'
-                              .format(vinit_req.status_code))
+        video_page = self.session.http.get(self.url, headers=dict(referer=self.url))
+        if video_page.status_code != 200:
+            raise PluginError('Could not get video page (HTTP Status {})'
+                              .format(video_page.status_code))
+
+        page_info_js = self._page_info.search(video_page.text)
+        if not page_info_js:
+            raise PluginError('Could not find page info')
 
-        video_status_js = self._video_status.search(vinit_req.text)
-        if not video_status_js:
-            raise PluginError('Could not find video status information!')
+        page_info = json.loads(page_info_js.group(1))
+        if not page_info['postDetail'].get('post'):
+            raise PluginError('Could not find video info')
 
-        video_status = json.loads(video_status_js.group(1))
+        video_json = page_info['postDetail']['post']['officialVideo']
 
-        if video_status['viewType'] == 'vod':
+        video_type = video_json['type']
+        if video_type == 'VOD':
             raise PluginError('VODs are not supported')
 
-        if 'liveStreamInfo' not in video_status:
-            raise PluginError('Stream is offline')
+        url_format, video_id = re.match(self._url_re, self.url).group('format', 'id')
+        if url_format == 'post':
+            video_id = str(video_json['videoSeq'])
+
+        video_status = video_json['status']
+        if video_status == 'ENDED':
+            raise PluginError('Stream has ended')
+        elif video_status != 'ON_AIR':
+            raise PluginError('Unknown video status: %s' % video_status)
+
+        stream_info_js = self.session.http.get(self._playinfo_url % video_id,
+                                               headers=dict(referer=self.url))
+        if stream_info_js.status_code != 200:
+            raise PluginError('Could not get stream info (HTTP Status {})'
+                              .format(stream_info_js.status_code))
 
-        stream_info = json.loads(video_status['liveStreamInfo'])
+        stream_info = json.loads(stream_info_js.text)
 
         streams = dict()
         # All "resolutions" have a variant playlist with only one entry, so just combine them
-        for i in stream_info['resolutions']:
-            res_streams = HLSStream.parse_variant_playlist(self.session, i['cdnUrl'])
+        for i in stream_info['result']['streamList']:
+            res_streams = HLSStream.parse_variant_playlist(self.session, i['serviceUrl'])
             if len(res_streams.values()) > 1:
                 log.warning('More than one stream in variant playlist, using first entry!')
 
-            streams[i['name']] = res_streams.popitem()[1]
+            streams[i['streamName']] = res_streams.popitem()[1]
```

Comment:
Just yielding a tuple of the stream name and HLSStream instance is cleaner than adding them to a dict and returning it afterwards, but that's just a "cosmetic" code suggestion.